tart2ms/fixvis.py

```diff
@@ -202,30 +202,33 @@ def synthesize_uvw(station_ECEF, time, a1, a2,
                     [padded_uvw, padded_time, padded_a1, padded_a2]))
 
 
-def rephase(vis, freq, pos, uvw, refdir, field_ids, phasesign=-1):
+def rephase(vis, freq, pos, uvw, refdir, field_ids, phasesign=-1, sel=None):
     """
         Rephasor operator
         -- rephases a field to a new phase centre
         freq - in Hz
         pos - tupple containing degree coordinates for RA and Dec for the epoch under consideration
         refdir - array of tuples with original field phase centres in RA and dec at the same epoch as pos (degrees), one per field
         phasesign - should be -1 for the NRAO baseline conventions
+        sel - (if not None same length as num rows) selects a portion of data in uvw, field_ids and vis
```

Comment:
It is necessary to optimize the graph to do the selection inside the function. I think the issue comes in that dask unnecessarily allocates many graph nodes with their own memory otherwise if done in the calling loop. This approach is orders of magnitude faster than the former.